src/vm/functions/conversions.rs

```diff
@@ -113,3 +114,91 @@ pub fn native_buff_to_uint_be(value: Value) -> Result<Value> {
     }
     return buff_to_int_generic(value, EndianDirection::BigEndian, convert_to_uint_be);
 }
+
+pub fn native_string_to_int_generic(
+    value: Value,
+    conversion_fn: fn(String) -> Result<Value>,
+) -> Result<Value> {
+    match value {
+        Value::Sequence(SequenceData::String(CharType::ASCII(ASCIIData { data }))) => {
+            match String::from_utf8(data) {
+                Ok(as_string) => conversion_fn(as_string),
+                Err(_error) => Ok(Value::none()),
+            }
+        }
+        Value::Sequence(SequenceData::String(CharType::UTF8(UTF8Data { data }))) => {
+            let flattened_bytes = data.into_iter().flatten().collect();
+            match String::from_utf8(flattened_bytes) {
+                Ok(as_string) => conversion_fn(as_string),
+                Err(_error) => Ok(Value::none()),
+            }
+        }
+        _ => Err(CheckErrors::UnionTypeValueError(
+            vec![
+                TypeSignature::max_string_ascii(),
+                TypeSignature::max_string_utf8(),
+            ],
+            value,
+        )
+        .into()),
+    }
+}
+
+fn safe_convert_string_to_int(raw_string: String) -> Result<Value> {
+    let possible_int = raw_string.parse::<i128>();
+    match possible_int {
+        Ok(val) => return Value::some(Value::Int(val)),
+        Err(_error) => return Ok(Value::none()),
+    }
+}
+
+pub fn native_string_to_int(value: Value) -> Result<Value> {
+    native_string_to_int_generic(value, safe_convert_string_to_int)
+}
+
+fn safe_convert_string_to_uint(raw_string: String) -> Result<Value> {
+    let possible_int = raw_string.parse::<u128>();
+    match possible_int {
+        Ok(val) => return Value::some(Value::UInt(val)),
+        Err(_error) => return Ok(Value::none()),
+    }
+}
+
+pub fn native_string_to_uint(value: Value) -> Result<Value> {
+    native_string_to_int_generic(value, safe_convert_string_to_uint)
+}
+
+pub fn native_int_to_string_generic(
+    value: Value,
+    conversion_fn: fn(bytes: Vec<u8>) -> Result<Value>,
```

Comment:
nit: I think it's a bit confusing that this conversion function expects a byte vector instead of an int, while the previous one (for string->int) expects a string.

Reply:
I added some comments to the functions that receive these functional inputs. Even though it's counter-intuitive, these are, I think, the most natural inputs.

In the case of the function that takes bytes, this is to interface with `Value::string_ascii_from_bytes` and `Value::string_utf8_from_bytes`. So, otherwise, I would need to introduce wrappers around these.